File: src/app/announcer.ts
import { SoundElement } from './sound-element';

export class Announcer {
  files: Array<string>;
  spritelist: object;
  constructor(_files: Array<string>, _spritelist: object) {
    this.files = _files;
    this.spritelist = _spritelist;
  }
}

File: lib/src/announcers/announcer.dart
import 'package:avalon_app/src/announcers/announcer_cathy.dart';

enum AnnouncerID { cathy, en_gb_c, en_gb_d }

abstract class Announcer {
  static const Map<AnnouncerID, Announcer> idMap = {
    AnnouncerID.cathy: AnnouncerCathy(),
    AnnouncerID.en_gb_c: AnnouncerEnGbC(),
    AnnouncerID.en_gb_d: AnnouncerEnGbD(),
  };
  final List<String> files = null;
  final Map<String, List<double>> spriteList = null;

  const Announcer();
}
